rllib/evaluation/env_runner_v2.py

```diff
@@ -1046,16 +1067,21 @@ def _try_find_policy_again(eval_data: AgentConnectorDataType):
                 # changed (mapping fn not staying constant within one episode).
                 policy: Policy = _try_find_policy_again(eval_data)
 
-            input_dict = _batch_inference_sample_batches(
-                [d.data.sample_batch for d in eval_data]
-            )
+            if policy.config.get("_enable_rl_module_api", False):
+                input_dict = concat_samples([d.data.sample_batch for d in eval_data])
+            else:
+                input_dict = _batch_inference_sample_batches(
```

Comment:
_batch_inference_sample_batches does nothing but concatenating AND setting SEQ_LENS to ones in the recurrent case. We don't need this because this is done implicitly.